libsplinter/src/service/network/interconnect/mod.rs

```diff
@@ -0,0 +1,719 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod error;
+mod scml;
+
+use std::collections::HashMap;
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::thread;
+
+use crate::matrix::{MatrixReceiver, MatrixRecvError, MatrixSender};
+use crate::network::dispatch::{
+    ConnectionId, DispatchLoopBuilder, DispatchLoopShutdownSignaler, DispatchMessageSender,
+    Dispatcher, MessageSender,
+};
+use crate::protos::component::{ComponentMessage, ComponentMessageType};
+
+pub use self::error::{ServiceInterconnectError, ServiceLookupError};
+
+/// The ServiceLookup trait provides an interface for looking up details about individual service
+/// connections.
+pub trait ServiceLookup: Send {
+    /// Retrieves the connection id for a given service id, if found.
```

Comment:
Is "if found" needed here?  (The  following Errors code clarifies that you get an error if it's not found. ) It's unclear,  gramatically, whether "if found" means if the connection ID is found or the service ID is found. Rather than trying to clarify, how about just deleting these two words?

This comment applies to similar lines elsewhere in this PR.

Reply:
The error is not about whether or not the service id exists, but rather if there's an unexpected error with the implementer of the trait.  I'll reword this.